### PR TITLE
Upgrade babel and bring babel runtime back

### PR DESCRIPTION
**Pull request by TrySound**
The problem you achieved was because of wrapped with runtime cjs helpers
and polyfills. In this diff I upgraded babel and added esm versions of
helpers. Polyfills are completely disabled which is the same behaviour
as inlined helpers. The size is increased because helpers in babel7 was
a bit changed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,15 +61,15 @@
...
     "rollup-plugin-commonjs": "^9.1.3",
     "rollup-plugin-node-resolve": "^3.3.0",
     "rollup-plugin-replace": "^2.0.0",
     "rollup-plugin-size-snapshot": "^0.4.1",
-    "sinon": "^5.0.0",
-    "size-limit": "^0.18.0"
+    "sinon": "^5.0.0"
   },
   "dependencies": {
+    "@babel/runtime": "^7.0.0-beta.47",
```
Comment (oliviertassinari) [inline package.json]: Would be better to match Material-UI's scope, beta 42 or something like that.

File: rollup.config.js [hunk 2]
```diff
@@ -14,9 +14,10 @@ const globals = {
   'prop-types': 'PropTypes',
 };
 
-const getBabelOptions = () => ({
+const getBabelOptions = ({ useESModules }) => ({
   exclude: '**/node_modules/**',
   runtimeHelpers: true,
+  plugins: [['@babel/transform-runtime', { polyfill: false, useBuiltIns: true, useESModules }]],
 });
```
Comment (oliviertassinari) [inline rollup.config.js]: Should we test that it's still working with all the browsers we support. Like IE 11, does it handle `Object.assign`?

Comment (TrySound) [reply rollup.config.js]: Test your latest release. It's the same.

File: package.json (hunk 1 above)
Comment (TrySound) [reply package.json]: Nope. Update to the latest version whenever possible. Small duplication in mismatched versions is ok. Will be fixed when babel7 will be released or mui will update Babel.

This is still better than inlining because the version of these plugins can be used in different package. So we still win.

**Comment by TrySound** [comment]
It's not the latest pr. I have another one improvement.

File: rollup.config.js (hunk 2 above)
Comment (oliviertassinari) [reply rollup.config.js]: How is that the same?
```js
var _Object$assign = _interopDefault(require('babel-runtime/core-js/object/assign'));
``` 
vs
```js
Object.assign
```

Comment (TrySound) [reply rollup.config.js]: Merge [this](https://github.com/oliviertassinari/react-event-listener/pull/74) nice PR to solve the problem.

**Comment by TrySound** [comment]
Rebased

**Comment by oliviertassinari** [comment]
### TL;DR

Sacrifize some bundle size to increase the overall deduplication potential.
```jsx
     path: 'dist/react-event-listener.cjs.js',
-    limit: '1.89 KB'
+    limit: '2.14 KB'
   }
```

**Comment by TrySound** [comment]
@oliviertassinari Why do you need size limit if size snapshot shows everything you need without manual patching config? Let's not use mismatched babel versions they can have breaking change in any time.

**Comment by oliviertassinari** [comment]
@TrySound For the same reason, I think that we should keep it for Material-UI. As we have been discussing it, I don't think that we can ask contributors to update the size snapshot for every single change. But we need to prevent bundle size regression. It's what size-limit does. If you work on a new feature and add another dependency you think is harmless, well size-limit will tell you.
TL;DR avoid regression with minimal overhead.

**Comment by TrySound** [comment]
Not every code change is regression. Feature can made your code bigger which is not BAD. But users should manually increase this number calculating shitty KB in mind. How much easier will be just run `yarn build`?

**Comment by TrySound** [comment]
This is small project. Even running on precommit is ok.

And still I don't get why asking contributors to run one command is bad. Want to contribute, provide all results of your contribution, probably write tests?
